tests/app/main/views/test_suppliers.py

```diff
@@ -1253,6 +1299,242 @@ def test_ccs_sourcing_role_can_not_update_supplier_name(self):
+                'frameworkInterest': [
+                    SupplierFrameworkStub(framework_slug="g-cloud-11", with_declaration=True).response()
+                ]
+            }
+        response = self.client.post(
+            '/admin/suppliers/1234/edit/registered-company-number',
+            data={'companies_house_number': '12345678'}
+        )
+        assert response.status_code == 302
+        assert response.location == 'http://localhost/admin/suppliers/1234'
+        self.data_api_client.update_supplier.assert_called_once_with(
+            1234,
+            {
+                'companiesHouseNumber': "12345678",
+                'otherCompanyRegistrationNumber': None,
+            },
+            "test@example.com"
+        )
+        assert self.data_api_client.update_supplier_declaration.call_args_list == ([
+            mock.call(
+                1234, 'g-cloud-11',
+                {'supplierCompanyRegistrationNumber': "12345678"}, "test@example.com"
+            )
+        ] if update_declaration else [])
+        self.assert_flashes("The details for ‘ABC’ have been updated.")
+
+    @pytest.mark.parametrize('update_declaration', [True, False])
+    def test_data_controller_role_can_update_other_registration_number(self, update_declaration):
+        self.user_role = 'admin-ccs-data-controller'
+        self.data_api_client.get_supplier.return_value = {
+            "suppliers": {"id": 1234, "otherCompanyRegistrationNumber": "abc123456", "name": "ABC"}
+        }
+        if update_declaration:
+            self.data_api_client.get_supplier_frameworks.return_value = {
+                'frameworkInterest': [
+                    SupplierFrameworkStub(framework_slug="g-cloud-11", with_declaration=True).response()
+                ]
+            }
+        response = self.client.post(
+            '/admin/suppliers/1234/edit/registered-company-number',
+            data={'other_company_registration_number': 'def98765'}
+        )
+        assert response.status_code == 302
+        assert response.location == 'http://localhost/admin/suppliers/1234'
+        self.data_api_client.update_supplier.assert_called_once_with(
+            1234,
+            {
+                'companiesHouseNumber': None,
+                'otherCompanyRegistrationNumber': "def98765",
+            },
+            "test@example.com"
+        )
+        assert self.data_api_client.update_supplier_declaration.call_args_list == ([
+            mock.call(
+                1234, 'g-cloud-11',
+                {'supplierCompanyRegistrationNumber': "def98765"}, "test@example.com"
+            )
+        ] if update_declaration else [])
+        self.assert_flashes("The details for ‘ABC’ have been updated.")
+
+    @pytest.mark.parametrize('update_declaration', [True, False])
+    def test_data_controller_role_can_change_companies_house_to_other_registration_number(self, update_declaration):
```

Comment:
Do we need a test for the reverse case (other registration number to companies house number)?

Reply:
I think we're probably fine for now.